File: src/lib/types/db/library/LibraryCache.ts
export interface LibraryCache {
  id: number;
  path: string;
  title: string | null;
  artist: string | null;
  album: string | null;
  album_artist: string | null;
  year: string | null;
  genre: string | null;
  track_number: number | null;
  disc_number: number | null;
  duration: number | null;
  bitrate: number | null;
  bits_per_sample: number | null;
  sample_rate: number | null;
  channels: number | null;
  audio_format: string | null;
  mime_type: string | null;
  file_size: number | null;
  extra_tags: string | null; // JSON string
  thumbnail_path: string | null;
  last_scanned_at: string | null;
}

export interface LibraryCacheCreate {
  path: string;
  title: string | null;
  artist: string | null;
  album: string | null;
  album_artist: string | null;
  year: string | null;
  genre: string | null;
  track_number: number | null;
  disc_number: number | null;
  duration: number | null;
  bitrate: number | null;
  bits_per_sample: number | null;
  sample_rate: number | null;
  channels: number | null;
  audio_format: string | null;
  mime_type: string | null;
  file_size: number | null;     // i64 côté Rust -> number côté JS
  extra_tags: string | null;    // JSON string
  thumbnail_path: string | null;
}

File: src-tauri/src/entity/library/library_cache.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct LibraryCache {
    pub id: i64,
    pub path: String,

    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub album_artist: Option<String>,

    pub year: Option<String>,
    pub genre: Option<String>,
    pub track_number: Option<i32>,
    pub disc_number: Option<i32>,

    pub duration: Option<f64>,
    pub bitrate: Option<i32>,
    pub bits_per_sample: Option<i32>,
    pub sample_rate: Option<i32>,
    pub channels: Option<i32>,

    pub audio_format: Option<String>,
    pub mime_type: Option<String>,

    pub file_size: Option<i64>,

    pub extra_tags: Option<String>, // JSON string
    pub thumbnail_path: Option<String>,

    pub last_scanned_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LibraryCacheCreate {
    pub path: String,

    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub album_artist: Option<String>,

    pub year: Option<String>,
    pub genre: Option<String>,
    pub track_number: Option<i32>,
    pub disc_number: Option<i32>,

    pub duration: Option<f64>,
    pub bitrate: Option<i32>,
    pub bits_per_sample: Option<i32>,
    pub sample_rate: Option<i32>,
    pub channels: Option<i32>,

    pub audio_format: Option<String>,
    pub mime_type: Option<String>,

    pub file_size: Option<i64>,

    pub extra_tags: Option<String>, // JSON sérialisé
    pub thumbnail_path: Option<String>,
}

